tests/test_rtls.py

```diff
@@ -2,23 +2,24 @@
 
 """Tests for `rtls` package."""
 
-import pytest
+from unittest.mock import patch
 
+from rtls.lib.git import GitLocator
+from rtls.lib.rpm import RpmLocator
+from rtls.lib.compose import ComposeLocator
+from rtls.lib.container import ContainerLocator
+from rtls.rtls import Locator
 
-from rtls import rtls
 
-
-@pytest.fixture
-def response():
-    """Sample pytest fixture.
-
-    See more at: http://doc.pytest.org/en/latest/fixture.html
+def test_locator_init():
     """
-    # import requests
-    # return requests.get('https://github.com/audreyr/cookiecutter-pypackage')
-
-
-def test_content(response):
-    """Sample pytest test function with the pytest fixture as an argument."""
-    # from bs4 import BeautifulSoup
-    # assert 'GitHub' in BeautifulSoup(response.content).title.string
+    GIVEN Locator initialized with a distro info path
+    WHEN the object is created
+    THEN the library objects are initialized
+    """
+    with patch('rtls.config.setup_distro_info'):
+        locator = Locator('https://example.com/git/example-full.yml')
+        assert type(locator.git) is GitLocator
+        assert type(locator.rpm) is RpmLocator
+        assert type(locator.compose) is ComposeLocator
+        assert type(locator.container) is ContainerLocator
```

Comment:
Hmm, now that I get to this point, I realize you are meaning to use 'Locator' as kind of a container for different types of locator objects.  I think most of what I said is still valid, but perhaps instead of a Factory to instantiate whatever type of locator is passed in, we do want some sort of 'LocatorWrapper' (not a real name, just trying to describe what I mean). This object seems to be intended as the entry point for interacting with the various locator subclasses.  It is a little hard to suggest a design here when I am not sure what has been discussed or envisioned for this process so far, but hopefully my suggestion at least give some possible directions to consider.